src/connectors/breadcrumb/connectBreadcrumb.js

```diff
@@ -0,0 +1,194 @@
+ */
+
+/**
+ * @typedef {Object} BreadcrumbRenderingOptions
+ * @property {function(item.value): string} createURL Creates an url for the next state for a clicked item. The special value `null` is used for the `Home` (or root) item of the breadcrumb and will return an empty array.
+ * @property {BreadcrumbItem[]} items Values to be rendered.
+ * @property {function(item.value)} refine Sets the path of the hierarchical filter and triggers a new search.
+ * @property {Object} widgetParams All original `CustomBreadcrumbWidgetOptions` forwarded to the `renderFn`.
+ */
+
+/**
+  * **Breadcrumb** connector provides the logic to build a custom widget
+  * that will give the user the ability to see the current path in a hierarchical facet.
+  *
+  * This is commonly used in websites that have a large amount of content organized in a hierarchical manner (usually e-commerce websites).
+  * @type {Connector}
+  * @param {function(BreadcrumbRenderingOptions, boolean)} renderFn Rendering function for the custom **Breadcrumb* widget.
+  * @return {function(CustomBreadcrumbWidgetOptions)} Re-usable widget factory for a custom **Breadcrumb** widget.
+  */
+export default function connectBreadcrumb(renderFn) {
+  checkRendering(renderFn, usage);
+  return (widgetParams = {}) => {
+    const { attributes, separator = ' > ', rootPath = null } = widgetParams;
+    const [hierarchicalFacetName] = attributes;
+
+    return {
+      getConfiguration: currentConfiguration => {
+        if (currentConfiguration.hierarchicalFacets) {
+          const facetSet = find(
+            currentConfiguration.hierarchicalFacets,
+            ({ name }) => name === hierarchicalFacetName
+          );
+          if (facetSet) {
+            if (
+              !isEqual(facetSet.attributes, attributes) ||
+              facetSet.separator !== separator
+            ) {
+              console.warn(
+                'Using Breadcrumb & HierarchicalMenu on the same facet with different options'
+              );
+            }
+            return {};
+          }
+        }
+
+        return {
+          hierarchicalFacets: [
+            {
+              attributes,
+              name: hierarchicalFacetName,
+              separator,
+              rootPath,
+            },
+          ],
+        };
+      },
+
+      init({ createURL, helper, instantSearchInstance }) {
+        this._createURL = facetValue => {
+          if (!facetValue) {
+            const breadcrumb = helper.getHierarchicalFacetBreadcrumb(
+              hierarchicalFacetName
+            );
+            if (breadcrumb.length > 0) {
+              return createURL(
+                helper.state.toggleRefinement(
+                  hierarchicalFacetName,
+                  breadcrumb[0]
+                )
+              );
+            } else {
```

Comment:
An function which does not explicitly return a value, actually returns `undefined`. So we don't need the last `else` block